integration/helpers/filesystem.ts

```diff
@@ -0,0 +1,45 @@
+import fse from "fs-extra";
+import stripIndent from "strip-indent";
+import path from "node:path";
+import url from "node:url";
+
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
+const root = path.resolve(__dirname, "../..");
+const TMP_DIR = path.join(root, ".tmp/integration");
+
+export async function writeTestFiles(
```

Comment:
This function, and the logic in `createTestDirectory` below were pulled from other files and just refactored into this file for reuse.